sycl/test-e2e/InvokeSimd/Regression/nbarrier_exec_in_order.cpp

```diff
@@ -157,14 +208,14 @@ bool test(QueueTY q) {
 
   bool passed = true;
   for (int i = 0; i < Size; i++) {
-    int etalon = i * 2 * Threads / Size;
-    if (etalon == Threads) // last stored chunk
-      etalon -= 1;
-    if (etalon > Threads) // excessive part of surface
-      etalon = 0;
-    if (out[i] != etalon) {
+    int ref = 2 * i * Threads / Size;
```

Comment:
this is the global buffer write you were talking about right? are we just doubling the offset to guarantee no overlap?

Reply:
I added a comment and changed the calculation slightly, but the resulting value remained the same.
TLDR: Reference value is the ID of the thread that stored it. To calculate it, we need to divide **_i_** by **_(VL / 2)_** and it comes from the fact that threads are writing their values with **_(VL / 2)_** offsets from each other.